spinta/dimensions/param/ufuncs.py

```diff
@@ -1,10 +1,140 @@
-from typing import Any
+from collections.abc import Generator
+from typing import Any, Iterator
 
-from spinta.core.ufuncs import Bind
+from spinta import commands
+from spinta.components import Model
+from spinta.core.ufuncs import Bind, Expr
 from spinta.core.ufuncs import Env
 from spinta.core.ufuncs import ufunc
+from spinta.datasets.components import Param
+from spinta.dimensions.param.components import ParamBuilder, ParamLoader
+from spinta.utils.schema import NotAvailable
+
+
+class _Loop:
+    expr: Expr
+    arg: Any
+
+    def __init__(self, expr, arg):
+        self.arg = arg
+        self.expr = expr
 
 
 @ufunc.resolver(Env, Bind)
 def param(env: Env, bind: Bind) -> Any:
     return env.params[bind.name]
+
+
+@ufunc.resolver(ParamBuilder, Expr)
+def select(env: ParamBuilder, expr: Expr) -> Any:
+    if isinstance(env.this, Model):
+        data = commands.getall(env.context, env.this, env.this.backend, query=expr, resolved_params=env.params)
+        return data
+
+
+@ufunc.resolver(ParamBuilder, Expr, name="getattr")
+def getattr_(env: ParamBuilder, expr: Expr):
+    args, kwargs = expr.resolve(env)
+    resolved = env.call("getattr", *args)
+    yield from resolved
+
+
+@ufunc.resolver(ParamBuilder, Generator, object, name="getattr")
+def getattr_(env: ParamBuilder, generator: Generator, obj: object):
+    for item in generator:
+        yield from env.call("getattr", item, obj)
+
+
+@ufunc.resolver(ParamBuilder, _Loop, Bind, name="getattr")
```

Comment:
This should work not only with `_Loop`, but with any kind of `GeneratorType`.

Reply:
Loop is no longer part of the implementation